overlord/snapstate/snapstate.go

```diff
@@ -1263,6 +1263,9 @@ func ensureInstallPreconditions(st *state.State, info *snap.Info, flags Flags, s
 	if err := checkDBusServiceConflicts(st, info); err != nil {
 		return flags, err
 	}
+	if err := checkDesktopFileIDsConflicts(st, info); err != nil {
+		return flags, err
+	}
```

Comment:
we should probably leave a TODO that if we implement a --disabled flag for install we should skip these checks

Reply:
Done